doc/design/OrionMongoSink.md

```diff
@@ -0,0 +1,110 @@
+#OrionMongoSink
+##Functionality
+`com.iot.telefonica.cygnus.sinks.OrionMongoSink`, or simply `OrionMongosink` is a sink designed to persist NGSI-like context data events within a MongoDB server. Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
+
+Independently of the data generator, NGSI context data is always transformed into internal Flume events at Cygnus sources thanks to `com.iot.telefonica.cygnus.handlers.OrionRestHandler`; click [here](from_ngsi_events_to_flume_events.md) to see the details. In the end, the information within these Flume events must be mapped into specific MongoDB data structures.
+
+###Mapping to MongoDB data structures
+MongoDB organizes the data in databases that contain collections of Json documents. Such organization is exploited by `OrionMongoSink` each time a Flume event is taken, by performing the following workflow:
+
+1. The bytes within the event's body are parsed and a `NotifyContextRequest` object container is created.
+2. A database called as the `fiware-service` header value within the event is created (if not existing yet).
+3. The context responses/entities within the container are iterated, and a collection is created (if not yet existing) for each unit data. The unit data depends on the chosen data model (see the configuration section):
+    * <i>collection-per-service-path</i>: the collection is called as the `fiware-servicePath` header value whitin the event.
+    * <i>collection-per-entity</i>: the collection is called as the concatenation of the `fiware-servicePath`_`destination` headers values within the event.
```

Comment:
Answering on order :)

1. The `destination` is really a header... but not in the HTTP notification but in the Flume event object ;) If you take a look on the paragraph before the section "Mapping to MongoDB data structures" you'll see a description about how HTTP notifications are converted into Flume events, which are the ones finally arriving to the sinks; such Flume events contain a `destination` header. I did not put all the details within this doc, but I pointed to the one giving all the details.
2. Storing the `servicePath` in the collection name mimics the desired behaviour for MySQL, where the table names where prefixed by that `servicePath`. Anyway, this is one of the three alternatives we are testing, in order to decide which one is the best in terms of efficiency, MongoDB usage, naming limitations, etc.
3. In the way of being fixed :)
4. We took into account those limitations you are mentioning in a wide internal discussion. That's way we decided to test the 3 alternatives (per-service-path, per-entity, per-attribute) and decide. Nevertheless, we did not heard about WiredTiger, so I'll encourage @gtorodelvalle to study it for the future. Thanks!